HashTable.java

```diff
@@ -0,0 +1,192 @@
+import java.util.*;
+import java.lang.reflect.Array;
+
+class HashTable <K, V> {
+
+    // The array cannot be smaller than MIN_CAPACITY at any time
+    private static final int MIN_CAPACITY = 32;
+    private int approxSize = Integer.highestOneBit(MIN_CAPACITY) << 1;
+    private int tableMask = approxSize - 1;
+
+    private HashEntry[] table;
+
+    // The current number of elements.
+    private int size;
+
+    // The current capacity of the array.
+    private int capacity;
+
+    // initialLoad determines how full the array should be made when resizing
+    private double initialLoad;
+    private double maxLoad;
+    private double minLoad;
+
+    public HashTable(){
+        size = 0;
+        capacity = MIN_CAPACITY;
+        maxLoad = 0.75;
+        minLoad = 0.25;
+        this.initialLoad = 0.5;
+
+        //HashEntry[] table = new HashEntry[capacity];
+        table = (HashEntry[]) Array.newInstance(HashEntry.class, capacity);
+    }
+
+    /**
+    * Returns the value associated with the specified key.
+    * @param key the key
+    * @return the value associated with <tt>key</tt>;
+    *         <tt>null</tt> if no such value
+    * @throws NullPointerException if <tt>key</tt> is <tt>null</tt>
+    */
+    public synchronized V get(K key){
+        if (key == null) throw new NullPointerException("first argument to get(K key) is null");
+
+        // Finds the key.
+        int index = hash(key) % capacity;
+        while(!key.equals(table[index].getKey())){
+            index = (index + 1) % capacity;
+        }
+        // Returns the value unless it is null.
```

Comment:
This comment doesn't add much value, it's just saying what the line of code does - remove it.